architecture.md

```diff
@@ -1,32 +0,0 @@
----
```

Comment:
This file is replaced by tidb-architecture.md and the alias has been added there. 

This might be related to https://github.com/pingcap/docs/pull/3300/ @ran-huang 
and https://github.com/pingcap/docs/pull/3094/files#r455682705 @lilin90 @uglyengineer